src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80DocValuesConsumer.java

```diff
@@ -115,6 +116,10 @@ private VectorTransfer getVectorTransfer(FieldInfo field) {
         if (VectorDataType.BINARY.getValue().equalsIgnoreCase(field.attributes().get(KNNConstants.VECTOR_DATA_TYPE_FIELD))) {
             return new VectorTransferByte(KNNSettings.getVectorStreamingMemoryLimit().getBytes());
         }
+        if (VectorDataType.BYTE.getValue().equalsIgnoreCase(field.attributes().get(KNNConstants.VECTOR_DATA_TYPE_FIELD))
```

Comment:
This function is getting overloaded with multiples ifs , Can we use Switch for more readability  and what if vectorDataType is null, Can we also check Null

```
private VectorTransfer getVectorTransfer(FieldInfo field) {
    String vectorDataType = field.attributes().get(KNNConstants.VECTOR_DATA_TYPE_FIELD);
    String knnEngine = field.attributes().get(KNNConstants.KNN_ENGINE);
    long memoryLimit = KNNSettings.getVectorStreamingMemoryLimit().getBytes();

    if (vectorDataType == null) {
        throw new IllegalArgumentException("VECTOR_DATA_TYPE_FIELD attribute is missing");
    }

    switch (vectorDataType.toLowerCase()) {
        case "binary":
            return new VectorTransferByte(memoryLimit);
        case "byte":
            if (KNNEngine.FAISS.getName().equalsIgnoreCase(knnEngine)) {
                return new VectorTransferByteToFloat(memoryLimit);
            }
        default:
            return new VectorTransferFloat(memoryLimit);
    }
}
```

Reply:
Refactored it to use switch case. But, I don't think we need a null check here because vectorDataType is an optional field which is by default set to float.